packages/fxa-auth-server/scripts/recorded-future/lib.ts

```diff
@@ -92,3 +95,102 @@ export const createHasTotp2faFn =
 
     return false;
   };
+
+export const createCredentialsLookupFn =
+  (client: ReturnType<typeof createClient<paths>>) =>
+  async (
+    logins: SearchResultIdentity[],
+    filter: components['schemas']['CredentialsLookupRequest']['filter']
+  ) => {
+    /**
+     * Note that there is an inconsistency/discrepancy between the API and the
+     * docs.  When a list of `subjects_login` is passed, all the results will
+     * be returned, even when there are over 500 items in the list.
+     *
+     * Since the input size is based on the results of the search results, it's
+     * unknown, so we'll chop up the list ourselves.
+     */
+
+    // each identity could have >1 credentials.  collect into an array of
+    // credentials and filter for cleartext secret
+    const credentialsWithCleartextSecret: components['schemas']['Credentials'][] =
+      [];
+    for (let i = 0; i < logins.length; i += defaultPerPageLimit) {
+      const subjectLogins = logins.slice(i, i + defaultPerPageLimit);
+      const { error, data } = await client.POST(
+        '/identity/credentials/lookup',
+        {
+          body: {
+            subjects_login: subjectLogins,
+            filter,
+          },
+        }
+      );
+
+      if (error) {
+        throw new Error(
+          `Recorded Future credentials lookup error: ${JSON.stringify(error)}`
+        );
+      }
+
+      for (const identity of data.identities || []) {
+        const cleartextSecretCreds = identity.credentials?.filter(
+          (id) => id.exposed_secret?.type === 'clear'
+        );
+
+        // the same combination of login and password could show up due to
+        // different leak sources
+        const creds = new Map();
+        cleartextSecretCreds?.forEach((x) =>
+          creds.set(
+            `${x.subject}${x.exposed_secret?.details?.clear_text_value}`,
```

Comment:
If x.subject and x.exposed_secret?.details?.clear_text_value are undefined, should they be added to the set. I'm guessing they aren't undefined in reality, but it's interesting the type indicates they could be.

Reply:
Yeah, that's to get the type checker stop complaining.  By this point, that value exists.